repo_autoindex/_impl/kickstart.py

```diff
@@ -0,0 +1,235 @@
+from typing import Optional, Type
+from collections.abc import AsyncGenerator
+import logging
+import configparser
+import json
+import os
+
+from .base import Repo, GeneratedIndex, Fetcher, IndexEntry, ICON_OPTICAL, ICON_QCOW
+from .template import TemplateContext
+from .tree import treeify
+from .yum import YumRepo
+
+LOG = logging.getLogger("repo-autoindex")
+
+
+class KickstartRepo(YumRepo):
```

Comment:
Due to lacking a proper definition of what a kickstart repo is, I could be wrong about this, but I suspect this code only works properly for kickstart repos satisfying certain conditions:

- only works if the kickstart repo contains exactly one yum repo
- only works if the yum repo is located at exactly `.`

Based on the content of `.treeinfo` it seems like these are only conventions, rather than something that all kickstart repos would abide by.

I don't have a problem with this if it works on all the repos we need it to, but it seems worthy of commenting on the limitations.